Tools/AP_Periph/AP_Periph.cpp

```diff
@@ -127,7 +127,9 @@ void AP_Periph_FW::init()
     logger.init(g.log_bitmask, log_structure, ARRAY_SIZE(log_structure));
 #endif
 
+#if AP_CHECK_FIRMWARE_ENABLED
```

Comment:
I think we should assert that AP_CHECK_FIRMWARE_ENABLED is enabled in AP_Periph. It is unsafe to have a peripheral without it. We need the crc checking or the user can brick their board

Reply:
why not an assert? periph just won't work properly without it